android/templates/build/AssetCryptImpl.java

```diff
@@ -1,93 +1,94 @@
-import android.os.Debug;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
 
 public class AssetCryptImpl implements KrollAssetHelper.AssetCrypt
 {
-	private static class Range
+	private static final String TAG = "AssetCryptImpl";
+
+	private static byte[] salt = {
+		<% for (let i = 0; i < salt.length - 1; i++){ -%>
+<%- '(byte)' + salt.readUInt8(i) + ', ' -%>
+<% } -%>
+<%- '(byte)' + salt.readUInt8(salt.length - 1) %>
+	};
+
+	private static final Collection<String> assets =
+		new ArrayList<String>(Arrays.asList(
+<% for (let i = 0; i < assets.length - 1; i++) { -%>
+			"Resources/<%- assets[i] %>",
+<% } -%>
+			"Resources/<%- assets[assets.length - 1] %>"
+		)
+	);
+
+	public AssetCryptImpl()
 	{
-		int offset;
-		int length;
-		public Range(int offset, int length)
-		{
-			this.offset = offset;
-			this.length = length;
+		try {
+			System.loadLibrary("ti.cloak");
+		} catch (Exception e) {
+			Log.e(TAG, "Could not load 'ti.cloak' library");
 		}
 	}
 
-	<%- encryptedAssets %>
-
 	@Override
 	public InputStream openAsset(String path)
 	{
-		byte[] bytes = fetchFilteredAssetBytes(path);
-		if (bytes == null) {
-			return null;
+		if (assets.contains(path)) {
+			byte[] bytes = getAssetBytes(path);
+			if (bytes == null) {
+				return null;
+			}
+			return new ByteArrayInputStream(bytes);
 		}
-		return new ByteArrayInputStream(bytes);
+		return null;
 	}
 
 	@Override
 	public String readAsset(String path)
 	{
-		byte[] bytes = fetchFilteredAssetBytes(path);
-		if (bytes == null) {
-			return null;
+		if (assets.contains(path)) {
+			byte[] bytes = getAssetBytes(path);
+			if (bytes == null) {
+				return null;
+			}
+			return new String(bytes);
 		}
-		return new String(bytes);
+		return null;
 	}
 
 	@Override
 	public Collection<String> getAssetPaths()
 	{
-		return assets.keySet();
-	}
-
-	private static byte[] fetchFilteredAssetBytes(String path)
-	{
-		TiApplication application = TiApplication.getInstance();
-		boolean isProduction = false;
-		if (application != null) {
-			isProduction = TiApplication.DEPLOY_TYPE_PRODUCTION.equals(application.getAppInfo().getDeployType());
-		}
-
-		if (isProduction && Debug.isDebuggerConnected()) {
-			Log.e("AssetCryptImpl", "Illegal State. Exit.");
-			System.exit(1);
-		}
-
-		Range range = assets.get(path);
-		if (range == null) {
-			return null;
-		}
-
-		return filterDataInRange(assetsBytes, range.offset, range.length);
+		return assets;
 	}
 
-	private static byte[] filterDataInRange(byte[] data, int offset, int length)
+	private static byte[] getAssetBytes(String path)
 	{
 		try {
-			Class clazz = Class.forName("org.appcelerator.titanium.TiVerify");
-			Method method = clazz.getMethod("filterDataInRange", new Class[] { data.getClass(), int.class, int.class });
-			return (byte[]) method.invoke(clazz, new Object[] { data, offset, length });
+			InputStream in = KrollAssetHelper.getAssetManager().open(path);
```

Comment:
Rather than read the full input stream into a byte array output stream, then doing a cipher on the byte array; why not use a CipherInputStream so it can stream through?

You're basically reading the full file into memory, decrypting, and returning those bytes - but for some calls you could return the InputStream unread but set up to read as needed (when called by `openAsset`) versus the fully read in version (when called by `readAsset`).